### PR TITLE
feat(providers/microsoft): add DefaultAzureCredential to data_lake

### PR DESCRIPTION
**Pull request by Lee-W**
This PR depends on https://github.com/apache/airflow/pull/33687

---
**^ Add meaningful description above**
Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/CONTRIBUTING.rst#pull-request-guidelines)** for more information.
In case of fundamental code changes, an Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvement+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in a newsfragment file, named `{pr_number}.significant.rst` or `{issue_number}.significant.rst`, in [newsfragments](https://github.com/apache/airflow/tree/main/newsfragments).

### CODE REVIEW

File: airflow/providers/microsoft/azure/hooks/data_lake.py [hunk 1]
```diff
@@ -308,18 +310,18 @@ def get_conn(self) -> DataLakeServiceClient:  # type: ignore[override]
...
             app_secret = conn.password
-            token_credential = ClientSecretCredential(tenant, app_id, app_secret)
-            return DataLakeServiceClient(
-                account_url=f"https://{conn.host}.dfs.core.windows.net", credential=token_credential, **extra
-            )
+            credential = ClientSecretCredential(tenant, app_id, app_secret)
+        elif conn.password:
+            credential = conn.password
+        else:
+            credential = DefaultAzureCredential()
```
Comment (phanikumv) [inline airflow/providers/microsoft/azure/hooks/data_lake.py]: Can you post a screenshot of a task working successfully with the `DefaultAzureCredential` that you introduced

Comment (Lee-W) [reply airflow/providers/microsoft/azure/hooks/data_lake.py]: Sure. I'll test and post a screenshot before I mark these PRs as read to review

**Comment by Lee-W** [comment]
圖片
圖片

**Review by phanikumv — COMMENTED** [review]
can you add example DAG for this change and unit tests as well

**Comment by Lee-W** [comment]
> can you add example DAG for this change and unit tests as well

I'm using the existing DAG for the testing, and I don't expect code changes to be needed on the DAG.  And yes, I'll add some unit tests

**Comment by phanikumv** [comment]
> > can you add example DAG for this change and unit tests as well
> 
> I'm using the existing DAG for the testing, and I don't expect code changes to be needed on the DAG. And yes, I'll add some unit tests

Does the existing DAG make use of DefaultAzureCredential?

**Comment by Lee-W** [comment]
> 



> > > can you add example DAG for this change and unit tests as well
> > 
> > 
> > I'm using the existing DAG for the testing, and I don't expect code changes to be needed on the DAG. And yes, I'll add some unit tests
> 
> Does the existing DAG make use of DefaultAzureCredential?

Whether to use `DefaultAzureCredential` is decided on the connection setup; if tenant and password are not set, it'll fallback to `DefaultAzureCredential`

**Comment by Lee-W** [comment]
Just add some unit tests

**Comment by potiuk** [comment]
@phanikumv -> are you good with it ?

**Review by potiuk — APPROVED** [review]

**Review by phanikumv — APPROVED** [review]

**Pull request merged**